app/models/nic/managed.rb

```diff
@@ -78,6 +95,46 @@ def dhcp_attrs
       }
     end
 
+    def copy_hostname_from_host
+      self.name = self.host.read_attribute :name
+    end
+
+    def set_provisioning_flag
+      if self.primary? && self.host && self.host.interfaces.detect(&:provision).nil?
+        self.provision = true
+      end
+    end
+
+    def update_lookup_value_fqdn_matchers
+      if self.primary? && self.fqdn_changed?
+        LookupValue.where(:match => "fqdn=#{fqdn_was}").update_all(:match => host.send(:lookup_value_match))
+      end
+    end
+
+    def drop_host_cache
+      if self.host
```

Comment:
Don't wrap the entire method in a condition, guard clauses make this easier to read. Try to 'unwrap' the ```set_provisioning``` and ``update_lookup_value_fqdn_matchers``` too please.
```ruby
def drop_host_cache
  return unless host.present?
  host.drop_primary_interface_cache if primary?
  host.drop_interface_cache               if provision? # you can use 'attribute?' syntax for boolean attributes
  true
end
```